### PR TITLE
publish page hook adding functionality

### PR DESCRIPTION
**Pull request by flundeen**
set up data structure for panel set and created branch adding funcitonality

### CODE REVIEW

File: src/app/comic/create/publish/BranchPlacer.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    
+    const startAdd = () => {
+        console.log("You have reached maximum amount of branches");
+        if(branCount < 3 ){
+            console.log("Start adding");
+            setAdding(true);
+        }
+    }
+
+    const addHook = (panel:number) => {
```
Comment (tstodter) [inline src/app/comic/create/publish/BranchPlacer.tsx:39]: Rename to addBranchHook, as Hooks are a react thing too.

File: src/app/comic/create/publish/BranchPlacer.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    const removeHook = () => {
+        let bholder = ps.branches;
+        for(let i = 0; i < 3; i++){
+            if(i == branCount){
+                bholder[i].panel = -1; //since not on a panel set's hook to -1
+                bholder[i].x = 0; //x-value of hook -----> TODO: align hook with mouse placement on panel
+                bholder[i].y = 0; //y-value of hook -----> TODO: align hook with mouse placement on panel
+                bholder[i].child_branch_uuid = uuidv4();
+                break;
```
Comment (tstodter) [inline src/app/comic/create/publish/BranchPlacer.tsx:79]: May need to call setPs again to persist these changes

Comment (flundeen) [reply src/app/comic/create/publish/BranchPlacer.tsx:79]: yes good catch i just fixed this

File: src/app/comic/create/publish/BranchPlacer.tsx (hunk 1 above)
Comment (flundeen) [reply src/app/comic/create/publish/BranchPlacer.tsx:39]: ah i didnt know that but i fixed it so addHook -> addBranchHook & removeHook -> removeBranchHook

**Pull request closed without merging**